src/NuGet.Indexing/NuGetIndexSearcher.cs

```diff
@@ -104,7 +104,7 @@ public static int TotalDownloadCounts(VersionResult versions)
         {
             int allVersions = TotalDownloadCounts(versions);
 
-            int thisVersion = versions.VersionDetails
+            int thisVersion = versions.AllVersionDetails
                 .Where(v => v.Version.Equals(version, StringComparison.OrdinalIgnoreCase))
```

Comment:
This is linear search... we should really have a Version to Download count dictionary.